gen/src/generate/char_property/bsearch_set.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::BTreeSet;
+
+/// A simple range-collapsing binary search array slice.
```

Comment:
I say keep it a closed range. The reason being border cases: since we're storing the characters bounding the range as `char` and not `u32`, if a boundary of the range is `LOW_SURROGATE - 1` or `MAX_UNICODE`, then one more for a half-open range would no longer be a valid character.